src/installer/corehost/CMakeLists.txt

```diff
@@ -9,6 +9,10 @@ if(MSVC)
 
     # Host components don't try to handle asynchronous exceptions
     add_compile_options(/EHsc)
+elseif (CMAKE_CXX_COMPILER_ID MATCHES GNU)
```

Comment:
I wonder if we could rather get rid of destructors on types used for statics. We've done that in coreclr long time ago as it was just causing troubles due to the fact that the order of destruction of such objects is undefined and if one ends up referencing the other or something similar, it can cause all kinds of funny issues.

Reply:
cc: @elinor-fung, @vitek-karas, @swaroop-sridhar

Reply:
I guess this issue can be addressed in a separate PR, since it affects all platforms.
I filed https://github.com/dotnet/runtime/issues/36580